### PR TITLE
AP_Periph: support output of MSP airspeed sensor

### PR DESCRIPTION
**Pull request by giacomo892**
Missing AP MSP side
INAV side: https://github.com/iNavFlight/inav/pull/6090

### CODE REVIEW

**Comment by giacomo892** [comment]
Branch for the ardupilot side:
https://github.com/giacomo892/ardupilot/tree/giacomo892_msp_airspeed

**Comment by magicrub** [comment]
ping @bugobliterator

**Comment by tridge** [comment]
@giacomo892 I did a small tidy up on this and rebased, then force pushed to your branch

**Comment by tridge** [comment]
I've added the AP_Airspeed driver too, based on your work with a few tweaks

**Comment by giacomo892** [comment]
@tridge looks cool, beside some stray files

**Comment by tridge** [comment]
> @tridge looks cool, beside some stray files

thanks, I've removed the stray commits

**Comment by tridge** [comment]
I've tested this successfully with a SDP3X airspeed sensor

**Comment by tridge** [comment]
this needs a new ARSPD_TYPE which probes for several types of sensors, so it is useful for people without CAN

**Comment by giacomo892** [comment]
> this needs a new ARSPD_TYPE which probes for several types of sensors, so it is useful for people without CAN

Cannot we enable MAVLINK on a serial port of the CAN modules so we can connect to them and change params easily?

**Comment by tridge** [comment]
> Cannot we enable MAVLINK on a serial port of the CAN modules so we can connect to them and change params easily?

possibly, but a bit tricky as GCS_MAVLink library links into so many things. It may be better to revive our old text menu system

**Comment by giacomo892** [comment]
> > Cannot we enable MAVLINK on a serial port of the CAN modules so we can connect to them and change params easily?
> 
> possibly, but a bit tricky as GCS_MAVLink library links into so many things. It may be better to revive our old text menu system

why not, it would be more than enough!

**Comment by giacomo892** [comment]
in msp.cpp
send_msp_airspeed 
airspeed.last_update_ms(); isn't giving us what we need, but rather the actual time we tried to read the sensor

in AP_Airspeed_MSP.cpp

in handle_msp
we use the temperature without any check 
sum_temp += pkt.temp*0.01;
we should check for INT16_MIN for invalid temperature

**Comment by tridge** [comment]
@giacomo892 I think the issues are fixed, can you re-test?

**Comment by tridge** [comment]
@giacomo892 I've tested MSP GPS and the 0x20 GPS health bit does change when I unplug

File: libraries/AP_Airspeed/AP_Airspeed_MSP.cpp [hunk 1]
```diff
@@ -50,6 +50,11 @@ void AP_Airspeed_MSP::handle_msp(const MSP::msp_airspeed_data_message_t &pkt)
         press_count /= 2;
     }
 
+    if (pkt.temp == INT16_MIN) {
```
Comment (giacomo892) [inline libraries/AP_Airspeed/AP_Airspeed_MSP.cpp:53]: This looks good if the airspeed sensor on the node has never sent any temperature update.
is it actually possible that a sensor still send the pressure but and invalid temperature?
in that case we will continue using the old stale temperature in the FC

Comment (tridge) [reply libraries/AP_Airspeed/AP_Airspeed_MSP.cpp:53]: once a sensor sends a temperature once then it will always send one with the pressure

**Pull request merged**